UI: Fix projectors not closing when source is deleted

cg2121 opened the pull request:
### Description
The projectors would still hold on to a reference of a source, even if the source was deleted, so use weak sources

### Motivation and Context
A source projector would still be opened, even when its source is deleted.

### How Has This Been Tested?
- Added source projector
- Deleted the source
- Made sure the projector was closed properly

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

jp9000 commented:
I don't think you need to have the remove signal changes. Just check to see if GetSource returns null and close the projector if that ends up being the case.

cg2121 commented:
It would benefit in other instances where the remove signal is used, as it only works for scenes not other sources.

tt2468 commented:
I had previously made a PR to call `source_remove` when a source is destructed, but after some discussion decided against trying to pursue it. My suggestion is to make sure the source is stored as a weak ref, then listen to the `source_destroy` signal to know when to close the projector.

cg2121 commented:
I've updated the PR to remove the source_remove signal change. I've changed the signal in the projector class to use 'destroy' instead of 'remove', per suggestion of @tt2468.

tt2468 commented on UI/window-projector.cpp:
```diff
@@ -17,11 +17,14 @@ static bool updatingMultiview = false, mouseSwitching, transitionOnDoubleClick;
+	OBSSource source = GetSource();
+
+	if (source)
+		removedSignal.Connect(obs_source_get_signal_handler(source),
```
Since we're using `destroy` instead of `remove`, we should probably change the name of this. Other than that, this looks good provided we don't get memory leaks.

cg2121 replied:
Updated the name of the function and the signal.

tt2468 commented on UI/window-projector.cpp:
```diff
@@ -371,7 +373,7 @@ void OBSProjector::UpdateProjectorTitle(QString name)
 OBSSource OBSProjector::GetSource()
 {
-	return source;
+	return OBSGetStrongRef(weakSource);
```
Wasn't able to find an answer myself. Does this increment the source ref when returning? Seems like it would increment the strong ref then return.

cg2121 replied:
No, it does not. No memory leaks occur. It is also done this way in media-controls.cpp. https://github.com/obsproject/obs-studio/blob/2ca0e7cc1af0314d4564f501b956fd58919ac30b/UI/media-controls.cpp#L286

RytoEX commented:
Merging based on approval above.
